afew/MailMover.py

```diff
@@ -106,7 +107,8 @@ def __update_db(self, maildir):
         Update the database after mail files have been moved in the filesystem.
         '''
         try:
-            check_call(['notmuch', 'new'])
+            if self.run_new:
+                check_call(['notmuch', 'new'])
```

Comment:
I'd prefer to do this check outside, everywhere where`__update_db` is called (similar to how dry_run is handled)